src/api/MetadataQuery.js

```diff
@@ -147,54 +158,84 @@ class MetadataQuery extends Base {
      * @param {Object} - metadata query response object
      * @return {Object} - flattened metadata query response object
      */
-    flattenMetdataQueryResponse = ({ entries, next_marker }: MetadataQueryResponse): FlattenedMetadataQueryResponse => {
+    flattenMetdataQueryResponse = (): FlattenedMetadataQueryResponse => {
+        const { entries, next_marker }: MetadataQueryResponse = this.metadataQueryResponse;
         return {
             items: entries.map<FlattenedMetadataQueryResponseEntry>(this.flattenResponseEntry),
             nextMarker: next_marker,
         };
     };
 
     /**
-     * @param {Object} response
+     * Success handler function storing flattened response in Cache
+     * @param {Object} templateSchemaResponse - metadata template schema response
      */
-    queryMetadataSuccessHandler = ({ data }: { data: MetadataQueryResponse }): void => {
+    successHandler = (templateSchemaResponse: ?MetadataTemplateSchemaResponse): void => {
+        this.metadataTemplate = getProp(templateSchemaResponse, 'data');
         const cache: APICache = this.getCache();
-        const filteredResponse = this.filterMetdataQueryResponse(data);
         // Flatten the filtered metadata query response and set it in cache
-        cache.set(this.key, this.flattenMetdataQueryResponse(filteredResponse));
+        cache.set(this.key, this.flattenMetdataQueryResponse());
         this.finish();
     };
 
+    /**
+     * Gets template schema
+     * @param {Object} response
+     */
+    getTemplateSchemaInfo = ({
+        data,
+    }: {
+        data: MetadataQueryResponse,
+    }): Promise<MetadataTemplateSchemaResponse | void> => {
+        this.metadataQueryResponse = this.filterMetdataQueryResponse(data);
+        const { entries } = data;
+
+        if (!entries || entries.length === 0) {
+            // if no items found for provided query, don't make API call to get template info
+            return Promise.resolve();
+        }
+
+        const metadata = getProp(entries, '[0].metadata');
+        this.templateScope = Object.keys(metadata)[0];
+        const instance = metadata[this.templateScope];
+        this.templateKey = Object.keys(instance)[0];
+
+        return this.metadataAPI.getSchemaByTemplateKey(this.templateKey);
```

Comment:
I think we should probably avoid having API classes call each others methods directly. Can the calling component handle this as part of its promise chain? Or should the `MetadataQuery` class be combined with or extend the `Metadata` class?

Reply:
There is not really a parent-child relationship between Metadata API and MetadataQuery API. So this entire logic merging metadata and template data type in `<ContentExplorer>` as this is the calling component. If we really don't want API classes calling each others methods (although there are examples of that currently), then I suggest adding the `flattenMetadata` util back in `src/utils` and `<ContentExplorer />` calling both APIs in its promise chain to gather the required data & types and merging & flattening using the`flattenMetadata` util.

Reply:
I think the main API class that calls other API classes is Feed, which has been pretty difficult to work with. This type of coordination would normally be done by a container invoking an asynchronous action, which most of our top-level Elements currently lack. If you're concerned about polluting ContentExplorer, you could consider writing a component-oriented helper class, similar to the [version history sidebar's](https://github.com/box/box-ui-elements/blob/master/src/elements/content-sidebar/versions/VersionsSidebarAPI.js).